src/utils/validators.js

```diff
@@ -0,0 +1,325 @@
+ */
+export function validateBalanceKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidBalanceKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'balance key',
+      fieldName
+    }))
+  }
+}
+
+/** Fields validators */
+
+/**
+ * Validate value to be a int64 amount.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {boolean} [allowZero] Is zero amount allowed.
+ * @param {number} [min] The minimum amount.
+ * @param {number} [max] The maximum amount.
+ *
+ * @throws {TypeError} Value should be a valid int64 amount.
+ */
+export function validateAmount ({
+  value,
+  fieldName = '',
+  allowZero = false,
+  max,
+  min
+}) {
+  if (!BaseOperation.isValidAmount(value, allowZero, max, min)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'should be a valid int64 amount',
+      params: { min, max },
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be an operation subject.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid operation subject.
+ */
+export function validateSubject ({ value, fieldName = '' }) {
+  if (!BaseOperation.isValidSubject(value)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'must be of type String 0-256 long',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be an asset code.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid asset code.
+ */
+export function validateAssetCode ({ value, fieldName = '' }) {
+  if (!BaseOperation.isValidAsset(value)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'must be valid asset code of type String 1-16 long',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a `xdr.FeeType` instance.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a `xdr.FeeType` instance.
+ */
+export function validateFeeType ({ value, fieldName = '' }) {
+  validateUndefined({ value, fieldName })
+
+  if (!(value instanceof xdr.FeeType)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      fieldName,
+      expectedType: 'xdr.FeeType'
+    }))
+  }
+}
+
+/**
+ * Validate value to be non-empty object with snake_cased keys.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be non-empty object with snake_cased keys.
+ */
+export function validateCreatorDetails ({ value, fieldName = '' }) {
+  const isCreatorDetailsValid = _isObject(value) &&
```

Comment:
redundant variable

Reply:
It's [Extract variable](https://refactoring.guru/extract-variable) refactoring. It makes conditional statement more understandable. Besides, it's easier to format code using a separate variable, than inside a conditional statement.

Reply:
@ardier16 no, it's not. the meaning of the expression is clear from the method's name. 

If function's goal was to do something else, that relies on the validity of details, then it would make sense. Like:

```
function processCreatorDetails (details) {
   const isValidDetails = a && b && c
   if (!isValidDetails) {
        throw ...
   }

   doSomethingWithDetails()
}

```

But all your function does is checking the condition, so variable to store this condition is redundant.

Reply:
but ok, you can leave it, too much words for such small problem